public/src/components/switchboard.tsx

```diff
@@ -97,61 +136,164 @@ const Switchboard: React.FC<InnerProps<SupportFrontendSwitches>> = ({
+          onClick={(): void => actionRemoveSwitchData(group, switchId, switchData.description)}
+          aria-label="removeSwitch"
+        >
+          <DeleteIcon />
+        </IconButton>
+      </div>
     );
   };
 
   const createGroupsFromData = (group: [string, SwitchGroup]): JSX.Element => {
     const [groupId, groupData] = group;
+    const { register, handleSubmit, errors } = useForm<FormData>();
+
+    const onSubmit = ({ switchId, description }: FormData): void => {
+      const updatedState = cloneDeep(data);
+
+      updatedState[groupId].switches[switchId] = {
+        description: description,
+        state: SwitchState.Off,
+      };
+      setData(updatedState);
+      setNewUnsavedData(true);
+      setPendingChange('Added', description, groupData.description);
+      setNeedToSaveDataWarning(true);
+    };
 
     return (
       <FormControl className={classes.formControl} key={groupId}>
-        <FormLabel>{groupData.description}</FormLabel>
+        <FormLabel>
+          {' '}
+          <strong>{groupData.description} </strong>
+        </FormLabel>
+        <br />
         {Object.entries(groupData.switches)
           .sort(sortByDescription)
           .map(([switchId, switchData]) =>
-            createSwitchesFromGroupData(switchId, switchData, groupId),
+            createSwitchesFromGroupData(switchId, switchData, group),
           )}
+        <div className={classes.inputGroup}>
+          <TextField
+            className={classes.input}
+            inputRef={register({
+              required: EMPTY_ERROR_HELPER_TEXT,
+              validate: switchId => {
+                return createDuplicateValidator(Object.keys(groupData.switches))(switchId);
+              },
+            })}
+            error={errors.switchId !== undefined}
+            helperText={errors.switchId ? errors.switchId.message : ''}
+            name="switchId"
+            label="Switch name"
+            margin="normal"
+            variant="outlined"
+            autoFocus
+            fullWidth
+          />
+          <span className={classes.inputSpacing} />
+          <TextField
+            className={classes.input}
+            inputRef={register({
+              required: EMPTY_ERROR_HELPER_TEXT,
+              validate: description => {
+                return createDuplicateValidator(
+                  Object.values(groupData.switches).map(switchData => switchData.description),
+                )(description);
+              },
+            })}
+            error={errors.description !== undefined}
+            helperText={errors.description ? errors.description.message : ''}
+            name="description"
+            label="Description"
+            margin="normal"
+            variant="outlined"
+            autoFocus
+            fullWidth
+          />
+          <span className={classes.inputSpacing} />
+          <div className={classes.buttons}>
+            <Button
+              variant="contained"
```

Comment:
The add button also doesn’t have any label, which we can add with aria-label (since we don’t have the text anywhere in the UI already):

```suggestion
              aria-label="Add switch"
              variant="contained"
```